### PR TITLE
feat: add EntityFilter component #637

### PR DESCRIPTION
**Pull request by deanex**
affects: @haulmont/jmix-front-generator, @haulmont/jmix-react-antd, test-puppeteer

### CODE REVIEW

File: packages/jmix-react-antd/src/ui/entity-filter/EntityFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  const [selectedProperty, setSelectedProperty] = useState<string>();
+  const [availableProperties, setAvailableProperties] = useState<string[]>(getAvailableProperties(metadata.entities, entityName, propertyNames));
+
+  const availablePropertiesOptions = availableProperties.map(propertyName => ({
+    label: <Msg entityName={entityName} propertyName={propertyName}/>,
+    value: propertyName,
+  }));
+
+  const onAddProperty = (addFilter: (defaultValue: JmixEntityFilter) => void) => {
+    addFilter({[selectedProperty!]: {_eq: ""}});
```
Comment (vyacheslav-pushkin) [inline packages/jmix-react-antd/src/ui/entity-filter/EntityFilter.tsx:34]: We are reusing a helper from `DataTable`. It is better to move `addFilter` to common utils (`src/util`).

**Review by vyacheslav-pushkin — CHANGES_REQUESTED** [review]

File: packages/jmix-react-antd/src/ui/entity-filter/EntityFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                {intl.formatMessage({id: "jmix.entityFilter.applyFilter"})}
+              </Button>
+            )}
+          </Space>
+        ))}
+      </Form.List>
+    </Form>
+  )
+}
+
```
[the remark this answers is not in the document]
Comment (deanex) [reply packages/jmix-react-antd/src/ui/entity-filter/EntityFilter.tsx:103]: I'll keep "Apply filter" button visible for user be able to reset filters if there is none

File: packages/jmix-react-antd/src/ui/entity-filter/EntityFilter.tsx (hunk 1 above)
Comment (deanex) [reply packages/jmix-react-antd/src/ui/entity-filter/EntityFilter.tsx:34]: `onAddProperty` receives `addFilter` callback for adding filters, but managing filters state in `EntityFilter` is done by [Form.List](https://ant.design/components/form/#Form.List). `addFilter` helper from `DataTable` isn't used here

File: packages/jmix-react-antd/src/ui/form/Form.tsx [hunk 3]
```diff
@@ -159,7 +159,7 @@ export type FormFieldProps = MainStoreInjected & {
 
 // forwardRef is required to avoid a console warning
 // as Form.Item will attempt to pass a ref
-export const FormField = injectMainStore(observer(React.forwardRef((props: FormFieldProps, _ref: any) => {
+export const  FormField = injectMainStore(observer(React.forwardRef((props: FormFieldProps, _ref: any) => {
```
[the remark this answers is not in the document]
Comment (deanex) [reply packages/jmix-react-antd/src/ui/form/Form.tsx]: Yep

**Review by vyacheslav-pushkin — APPROVED** [review]

**Pull request merged**